pkg/kuttlctl/cmd/root.go

```diff
@@ -34,6 +37,7 @@ and serves as an API aggregation layer.
 	cmd.AddCommand(newErrorsCmd())
 	cmd.AddCommand(newTestCmd())
 	cmd.AddCommand(newVersionCmd())
+	configFlags.AddFlags(cmd.PersistentFlags())
```

Comment:
With this the following is added to the default command

```
❯ bin/kubectl-kuttl --help
...
Flags:
      --as string                      Username to impersonate for the operation. User could be a regular user or a service account in a namespace.
      --as-group stringArray           Group to impersonate for the operation, this flag can be repeated to specify multiple groups.
      --as-uid string                  UID to impersonate for the operation.
      --cache-dir string               Default cache directory (default "/Users/lcx/.kube/cache")
      --certificate-authority string   Path to a cert file for the certificate authority
      --client-certificate string      Path to a client certificate file for TLS
      --client-key string              Path to a client key file for TLS
      --cluster string                 The name of the kubeconfig cluster to use
      --context string                 The name of the kubeconfig context to use
      --disable-compression            If true, opt-out of response compression for all requests to the server
  -h, --help                           help for kubectl-kuttl
      --insecure-skip-tls-verify       If true, the server's certificate will not be checked for validity. This will make your HTTPS connections insecure
      --kubeconfig string              Path to the kubeconfig file to use for CLI requests.
  -n, --namespace string               If present, the namespace scope for this CLI request
      --request-timeout string         The length of time to wait before giving up on a single server request. Non-zero values should contain a corresponding time unit (e.g. 1s, 2m, 3h). A value of zero means don't timeout requests. (default "0")
  -s, --server string                  The address and port of the Kubernetes API server
      --tls-server-name string         Server name to use for server certificate validation. If it is not provided, the hostname used to contact the server is used
      --token string                   Bearer token for authentication to the API server
      --user string                    The name of the kubeconfig user to use

Use "kubectl-kuttl [command] --help" for more information about a command.
...
```

A rest client config can be generated from these flags https://pkg.go.dev/k8s.io/cli-runtime/pkg/genericclioptions#ConfigFlags.ToRESTConfig

Reply:
Hmm, most od these seem specific enough. It would be great to achieve this level of consistency with `kubectl`.

However seeing `namespace` there got me thinking... `kuttl test` subcommand at least [already has a namespace flag](https://github.com/kudobuilder/kuttl/blob/762b0f9d2a6d64816ba80bf41b39d97fbcf8c215/pkg/kuttlctl/cmd/test.go#L260). I wonder how the two definitions would interact... 🤔 

And related to that: how do we make sure nothing breaks (or at least we are able to notice before releasing) in the future when we pick up a new version of the library which adds a new flag which we already happen to be using for another purpose 😟 

Thoughts?

Reply:
Ah, if I understand it correctly, this library [just copies](https://github.com/kubernetes/cli-runtime/blob/dd30b60637c4ce4981151982b7437d63663da7ec/pkg/genericclioptions/config_flags.go#L181) the value of the flag into the rest config. If it's that simple, then maybe we should do just that for the three `--as*` flags, to be on the safe side?

Reply:
Yeah I agree adding all of the flags from `genericclioptions` might be too wide a scope for this change and have the potential to collide with kuttl functionality so would take more time and care to implement along with test. 

> Ah, if I understand it correctly, this library [just copies](https://github.com/kubernetes/cli-runtime/blob/dd30b60637c4ce4981151982b7437d63663da7ec/pkg/genericclioptions/config_flags.go#L181) the value of the flag into the rest config. The heavy lifting is done by client-go? If it's that simple, then maybe we should do just that for the three `--as*` flags, to be on the safe side?

Lets do it! I will take this PR in the direction of adding the three `--as*` flags and copy the value of the flag if set into the rest config 🥳